From fe5f3cb9283f5067a3e8b327e8647d07b83266c5 Mon Sep 17 00:00:00 2001
From: Hains van den Bosch <hainsvdbosch@ziggo.nl>
Date: Sun, 1 Apr 2018 11:38:14 +0200
Subject: [PATCH] 029-move-field-atomic-flags-to-the-bottom-of-structure

Adding a field of datatype 'unsigned long' in the middle between the other 'tasks states' is too big
and causes a kernel panic.
Field 'atomic_flags' may have a datatype of 'short' but not int, long or long long.

Due to the 001-linux-dreambox-kernel-patch, the 'atomic_flags' field
must be declared after the 'struct task_cputime cputime_expires' field.

This causing no kernel panic and the box boot flawlessly.

---
 include/linux/sched.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/sched.h b/include/linux/sched.h
index 2d4ab76..074b33b 100644
--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -1313,8 +1313,6 @@ struct task_struct {
 	unsigned sched_reset_on_fork:1;
 	unsigned sched_contributes_to_load:1;
 
-	unsigned long atomic_flags; /* Flags needing atomic access. */
-
 	pid_t pid;
 	pid_t tgid;
 
@@ -1367,6 +1365,8 @@ struct task_struct {
 	struct task_cputime cputime_expires;
 	struct list_head cpu_timers[3];
 
+	unsigned long atomic_flags; /* Flags needing atomic access. */
+
 /* process credentials */
 	const struct cred __rcu *real_cred; /* objective and real subjective task
 					 * credentials (COW) */
-- 
2.15.1

